Core/include/Acts/TrackFitting/GlobalChiSquareFitter.hpp

```diff
@@ -58,7 +58,7 @@ struct Gx2FitterExtensions {
       Delegate<void(const GeometryContext&, const CalibrationContext&,
                     const SourceLink&, TrackStateProxy)>;
 
-  using Updater = Delegate<Result<void>(const GeometryContext&, TrackStateProxy,
+  using Updater = Delegate<Result<void>(const GeometryContext&, const Surface*, TrackStateProxy,
```

Comment:
Can you not get the surface from the track state?

Reply:
What do you think about this one @beomki-yeo? I think it's preferable to not pass in the surface explicitly, if you can get it from the track state.